source/_posts/2016-11-08-Vapor-mysql.md

```diff
@@ -0,0 +1,195 @@
+---
+title: Adding MySQL to your Vapor project
+date: 2016-11-08 21:31:21
+tags: vapor,mysql
+authorId: olha
+categories:
+- Vapor
+---
+
+In this tutorial we will walk through how to add a database layer to a [Vapor](vapor.codes) project. We will be using MySQL and will assume that you have set it up on your local machine [(guide here)](https://dev.mysql.com/doc/refman/5.6/en/osx-installation-pkg.html).
+
+We are continuing on our 'Car' project. You don't necacery have to have read that tutorial if you already are a Vapor rock star. If not then please find it [here](https://engineering.nodesagency.com/articles/Vapor/6-simple-steps-to-setup-vapor/).
+
+### 1. Adding the MySQL package
+
+Open your Vapor project and open the `Package.swift` file and add the extra dependecy under the main vapor dependency:
+
+```swift
+import PackageDescription
+
+let package = Package(
+    name: "Drive",
+    dependencies: [
+        .Package(url: "https://github.com/vapor/vapor.git", majorVersion: 1, minor: 1),
+        
+        //Package for MySQL provider for Vapor
+        .Package(url: "https://github.com/vapor/mysql-provider", majorVersion: 1, minor: 0)
+    ],
+    exclude: [
+        "Config",
+        "Database",
+        "Localization",
+        "Public",
+        "Resources",
+        "Tests",
+    ]
+)
+```
+Now open your terminal (make sure you are cd into the project) and run `vapor build --mysql` and after that `vapor xcode --mysql`, this will make sure everything is compiled correctly with Vapor and linked with MySQL on your local machine. 
+
+You should now be able to see some of the new dependencies in the `Dependencies` folder in your project. (MySQL, VaporMySQL, FluentMySQL etc).
+
+### 2. Setting up the DropLet to support the provider
+
+Now open your main.swift file and add the following lines of code:
+
+```swift
+...
+//Importing our new MySQL provider
+import VaporMySQL
+
+let drop = Droplet()
+
+// Add providers. This tells Vapor that we are using the VaporMySQL provider, this will bind the data to the database and the models automaticlly down the line
+try drop.addProvider(VaporMySQL.Provider.self)
+
+//Making sure that Vapor runs our migrations / preperations for our model(s) 
+drop.preparations.append(Car.self)
+
+....
+
+drop.run()
+```
+Now lets jump into our `Car.swift` file (and down in the extension of the model) add some actual Preparation code. 
```

Comment:
- s/Preparation/preparation
- also mentioning that preparation is something what people might now as migration in other framework might be nice